src/riscv-hybrid-integration.adoc

```diff
@@ -279,9 +279,20 @@ shown in xref:default-csrnames-added[xrefstyle=short].
 
 {cheri_default_ext_name} optionally allows <<MODESW>> to execute in debug mode.
 
-When entering debug mode, the core always enters {cheri_cap_mode_name}.
+When entering debug mode, whether the core enters {cheri_int_mode_name} or
+{cheri_cap_mode_name} is UNSPECIFIED.
```

Comment:
Couldn't we just say that the CHERI execution mode in debug is defined, for example, through a CRE bit or through the mode bit in `dinfc`?

Reply:
I often find that unspecified features are tricky to handle and give rise to problems in the long term. We do not need to add extra state. We could simply:
* Define that the CHERI execution mode is given by the mode bit in `dinfc`
* Clearly state that CSR accesses to dinfc are CLEN wide regardless of CHERI execution mode

If we like the unspecified part, we could indicate that the initial value of `dinfc.mode` is implementation-defined and that implementations may make `dinfc` read-only meaning that the CHERI mode in debug cannot be changed.

I guess the only awkward bit with this proposal is what happens when the implementation does assign pcs to the program buffer and/or it is possible to execute MODESW in debug mode. Perhaps `dinfc` just reflects the current mode?

Reply:
Its minor, but the specification says this in reference to CLEN access for new CHERI CSRs:
> Also, Zcherihybrid adds the new capability CSRs listed in Table 28.

Table 28 does not list dinfc.

>dinfc.M reports the mode already

I don't think it does. We removed that feature in this commit which is already merged into `main`:

https://github.com/riscv/riscv-cheri/commit/74f05ec206e5152abb904e9b20460bd34856a642#diff-ea4440c312373a6c9e166a66f393ba1c7025400a239982f21123dbf15d06fbff

Reply:
ok so the table needs fixing
the reset value of dinfc isn't a concern - what we need is for the mode to be updated on the entry into debug mode which is currently missing

Reply:
the current spec is fine as dinfc is a new CLEN CSR
<img width="674" alt="image" src="https://github.com/user-attachments/assets/967a2e2c-ded7-441f-8d2f-812f1dbdc908">

Reply:
done